### PR TITLE
feat(medusa, medusa-plugin-brightpearl): Inventory management for Brightpearl

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add multi warehouse support to brightpearl plugin

Fixes CORE-1013

### CODE REVIEW

File: packages/inventory/src/models/index.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
-export * from "./reservation-item"
+export {ReservationItem} from "./reservation-item"
```
Comment (olivermrbl) [inline packages/inventory/src/models/index.ts:1]: todo: Linting issue

File: packages/inventory/src/models/reservation-item.ts [hunk 2]
```diff
@@ -18,6 +24,9 @@ export class ReservationItem extends SoftDeletableEntity {
   @Column()
   quantity: number
 
+  @DbAwareColumn({ type: "enum", enum: ReservationType, default: ReservationType.INTERNAL })
+  type: ReservationType
+
```
Comment (olivermrbl) [inline packages/inventory/src/models/reservation-item.ts:29]: question: Curious, what spawned the need for this property?

**Review by olivermrbl — COMMENTED** [review]
@srindom Will you give this a review when time allows for it? Think you are best suited regarding the Brightpearl changes :)

**Comment by olivermrbl** [comment]
@pKorsholm Please add a changeset :)

File: packages/inventory/src/models/reservation-item.ts (hunk 2 above)
Comment (srindom) [inline packages/inventory/src/models/reservation-item.ts:29]: thought: would love if we could avoid this - looks like it is not currently being used?

I would be fine with an "external_id" prop if that would be helpful.

**Review by srindom — COMMENTED** [review]
I find it a bit hard to follow the PR - can you add some context about which flows have been added, modified and what would be your suggestion for how we review :)

File: packages/medusa/src/services/product-variant-inventory.ts [hunk 3]
```diff
@@ -396,6 +410,19 @@ class ProductVariantInventoryService extends TransactionBaseService {
...
       })
     )
+
+    // emit event
+    await Promise.all(
+      reservationItems.map(async (reservationItem) => {
+        await this.eventBusService_
+          .withTransaction(this.activeManager_)
+          .emit(ProductVariantInventoryService.Events.RESERVATION_CREATED, {
+            reservationItemId: reservationItem.id,
```
Comment (srindom) [inline packages/medusa/src/services/product-variant-inventory.ts]: question: don't we usually snake_case these?

Comment (srindom) [reply packages/medusa/src/services/product-variant-inventory.ts]: also why is this necessary when we already have a `reservation-item.created` event in the inventory module?

**Review by olivermrbl — APPROVED** [review]
LGTM, though I haven't reviewed the Brightpearl service as I am not familiar with the system. Would like @srindom to lead these efforts.

That said, honestly, I am fine pushing for a merge now and resolving issues related to BP in follow-up PRs. The important thing right now is that we put a halt to the snapshot acrobatics, that we've been performing the last couple of days 😅 

But this is great work @pKorsholm 💯

**Comment by pKorsholm** [comment]
> snapshot acrobatics

would be great to move past that indeed 😂

**Review by srindom — APPROVED** [review]
LGTM!
